storage/block/block.go

```diff
@@ -335,32 +336,34 @@ func (dbb *databaseSeriesBlocks) MaxTime() time.Time {
 }
 
 func (dbb *databaseSeriesBlocks) BlockAt(t time.Time) (DatabaseBlock, bool) {
-	b, ok := dbb.elems[t]
+	b, ok := dbb.elems[m3dbTime.NewUnixNano(t)]
 	return b, ok
 }
 
-func (dbb *databaseSeriesBlocks) AllBlocks() map[time.Time]DatabaseBlock {
+func (dbb *databaseSeriesBlocks) AllBlocks() map[m3dbTime.UnixNano]DatabaseBlock {
 	return dbb.elems
 }
 
 func (dbb *databaseSeriesBlocks) RemoveBlockAt(t time.Time) {
-	if _, exists := dbb.elems[t]; !exists {
+	tNano := m3dbTime.NewUnixNano(t)
+	if _, exists := dbb.elems[tNano]; !exists {
 		return
 	}
-	delete(dbb.elems, t)
+	delete(dbb.elems, tNano)
 	if !dbb.min.Equal(t) && !dbb.max.Equal(t) {
 		return
 	}
 	dbb.min, dbb.max = timeZero, timeZero
 	if len(dbb.elems) == 0 {
 		return
 	}
-	for k := range dbb.elems {
-		if dbb.min == timeZero || dbb.min.After(k) {
-			dbb.min = k
+	for key := range dbb.elems {
+		keyTime := key.Time()
+		if dbb.min == timeZero || dbb.min.After(keyTime) {
```

Comment:
Might make sense to convert `dbb.min` and `dbb.max` to use `UnixNano` as well?

Reply:
I can change this, but I think using the existing time.Time interface is nicer just because people are familiar with it.

Reply:
Basically my rationale is that in this case the comparison are fairly simple, but in other cases as I make these changes there will be things that will be using more complicated functions like Truncate() and Round() and it will just keep everything cleaner if I only do the conversions when putting things in and pulling things out of the map